app/src/conditionals.ts

```
@@ -169,3 +169,230 @@ registerCompiler('and', (expression: ConditionalExpression) => {
     return () => true;
   }
 });
+
+/**
+ * Trims whitespace in order to sanitize comparisons
+ * @param input The input string to sanitize
+ * @returns The output with whitespace trimmed
+ */
+const sanitizeComparisonInput = (input: string): string => {
+  return input.trim();
+};
+
+/**
+ * Checks if the item is a string array
+ * @param value The value to check
+ * @returns True iff is an array of strings
+ */
+const isStringArray = (value: unknown): value is string[] => 
+  Array.isArray(value) && value.every(item => typeof item === 'string');
+
+/**
+ * This condition checks that the list contains the targeted entry. If an error
+ * occurs or other edge case, returns false to reflect 'not' containing.
+ */
+registerCompiler('contains', (expression: ConditionalExpression) => {
```

Everything looks great overall.
I notice that the` contains` and `does-not-contain` function currently have error handling that silently returnss false or true when unexpected inputs are present.
Will it be helpful to log or throw an error may be for example when non array is passed as value? just thinking it through for production env.

---

As discussed in standup - this would be better handled at a higher level. Right now you can put this condition on any data type. If the user does this, throwing an error results in lots and lots of errors in the console which will get reported in bugsnag. I have silently failed for this reason.